### ISSUE TITLE
Use flat arrays for Colors and Positions when CanvasKit makes them available

### ISSUE THREAD

**Issue by harryterkelsen** [flutter/flutter#58824]
Currently to pass a list of Colors or a List of positions to CanvasKit, you need to instantiate a 2D JavaScript array, only for that array to be flattened and copied into WASM memory by CanvasKit. It would be faster if we could give CanvasKit a pre-flattened array.

**Comment by hterkelsen** [comment]
@kjlubick @yjbanov

**Comment by MichealReed** [comment]
Landed per CanvasKit 0.17.0 --- 

```
In all places where color arrays are accepted (gradient makers, drawAtlas, and MakeSkVertices), You can now provide either flat Float32Arrays of float colors, Uint32Arrays of int colors, or 2d Arrays of Float32Array(4) colors. The one thing you should not pass is an Array of numbers, since canvaskit wouldn't be able to tell whether they're ints or floats without checking them all. The fastest choice for gradients is the flat Float32Array, the fastest choice for drawAtlas and MakeSkVertices is the flat Uint32Array.
```

**Comment by harryterkelsen** [comment]
We do this now.
